Source/OpenDRIVE/Private/OpenDRIVEActor.cpp

```diff
@@ -0,0 +1,108 @@
+#include "OpenDRIVEActor.h"
+#include "OpenDRIVE.h"
+#include "Misc/UObjectToken.h"
+#include "Logging/MessageLog.h"
+#include "Roadmanager.hpp"
+#include "Kismet/GameplayStatics.h" 
+#if WITH_EDITOR
+#include "Editor.h"
+#include "Editor/EditorEngine.h"
+#include "Subsystems/ImportSubsystem.h"
+#endif
+ 
+// Sets default values
+AOpenDRIVEActor::AOpenDRIVEActor()
+{
+	// Set this actor to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
+	PrimaryActorTick.bCanEverTick = false;
+
+}
+
+// Called when the game starts or when spawned
+void AOpenDRIVEActor::BeginPlay()
+{
+	Super::BeginPlay();
+	LoadOpenDrive();
+}
+
+#if WITH_EDITOR 
+void AOpenDRIVEActor::PostEditChangeProperty(FPropertyChangedEvent& PropertyChangedEvent)
+{
+	Super::PostEditChangeProperty(PropertyChangedEvent);
+
+	FName PropertyName = (PropertyChangedEvent.MemberProperty != NULL) ? PropertyChangedEvent.MemberProperty->GetFName() : NAME_None;
+
+	if (PropertyName == GET_MEMBER_NAME_CHECKED(AOpenDRIVEActor, OpenDriveAsset))
+	{
+		LoadOpenDrive();
+	}
+}
+
+void AOpenDRIVEActor::CheckForErrors()
+{
+	Super::CheckForErrors();
+	FMessageLog MapCheck("MapCheck");
+
+	if (!GetLevel()) return;
+	if (GetLevel()->IsPersistentLevel()) return;
+	if (OpenDriveAsset) {
+		MapCheck.Warning()
+			->AddToken(FUObjectToken::Create(this))
+			->AddToken(FTextToken::Create(FText::FromString("is a streamed level but has an OpenDRIVE set, its OpenDRIVE will be ignored")));
+	}
+}
+#endif
+
+void AOpenDRIVEActor::LoadOpenDrive()
+{
+	if (!OpenDriveAsset) return;
+	if (GetLevel() && !GetLevel()->IsPersistentLevel()) {
+		UE_LOG(OpenDriveLog, Warning, TEXT("%s is a streamed level but has an OpenDRIVE set, its OpenDRIVE will be ignored"), *(GetFName().ToString()));
+		return;
+	}
+	if (!roadmanager::Position::GetOpenDrive()->LoadOpenDriveContent(TCHAR_TO_UTF8(*(OpenDriveAsset->XodrContent)))) {
+		UE_LOG(OpenDriveLog, Error, TEXT("Failed to load OpenDRIVE asset %s"), *(OpenDriveAsset->GetFName().ToString()));
+		return;
+	}
+#if WITH_EDITOR
+	if (GEditor && !bRegisteredReimportCallback) {
+		GEditor->GetEditorSubsystem<UImportSubsystem>()->OnAssetReimport.AddUObject(this, &AOpenDRIVEActor::OnObjectReimported);
+		bRegisteredReimportCallback = true;
+	}
+#endif
+}
+
+#if WITH_EDITOR
+void AOpenDRIVEActor::OnObjectReimported(UObject* InObject)
+{
+	if (InObject == OpenDriveAsset)
+	{
+		LoadOpenDrive();
+	}
+}
+
+void AOpenDRIVEActor::PostEditMove(bool bFinished)
+{
+	TArray<AActor*> openDRIVEActorInScene;
```

Comment:
> Let's say developer A adds the actor in a level unbeknownst to developer B. When developer B then tries to add the actor, it just disappear?! "WTH?!"

If I've got it correctly, when A drops an OpenDRIVEAsset in the scene (which creates an actor), if B wants to drop one too, B's OpenDRIVEActor will get destroy right ? 
Because this is not what's happening here : dropping a new OpenDRIVEAsset in the scene destroys the previous one (so A's one in your example). 

So basically, to load your OpenDRIVEAsset, you have two possibilities : 
1. Drag and drop a new one in the scene, and the previous OpenDRIVEActor (and Asset) will be replaced.
2.  OR if there's already an OpenDRIVEActor in the scene, you can always change the OpenDRIVEAsset in the actor's detail panel (like in the WorldSettings).

I think the drag and drop option is more user friendly than the second option and the current one (WorldSettings).
But of course we can still look for a better solution.

Reply:
> Because this is not what's happening here : dropping a new OpenDRIVEAsset in the scene destroys the previous one (so A's one in your example).

I still think that's an issue. What if developer B dropped the asset by accident? It deletes the previous one without much warning about it.

I don't think we should destroy anything automatically, especially things that were user-created. That's just bad behavior. If the user wants to have N actors in the scene, it's not our place to decide which ones will be deleted and which one will be the lucky winner that gets kept. What we could do is 
* warn the user that the current setup (multiple actors) is not handled and action should be taken
* prevent any further `LoadOpendrive` call until this issue is resolved

That way, there's no destroying on our part, the end user clean things up the way they want to. And we don't have to jump through hoops trying to ensure that there's NO WAY to create two actors: you can create as much as you want, but we'll explicitly warn you that things won't work until you clean it up.

Reply:
I made a quick fix : I'm displaying a basic debug message when a second actor is created, and locking every LoadOpenDrive() calls by checking if there's multiple OpenDRIVEActor. 
Once the error fixed, it will load the OpenDrive from the remaining actor.

Reply:
(and I also noticed that the actor was unmovable, but it's fixed now)